lib/spack/spack/test/cc.py

```diff
@@ -390,6 +437,42 @@ def test_ccld_with_system_dirs():
             test_args_without_paths)
 
 
+def test_ccld_with_system_dirs_isystem():
```

Comment:
I think it's worth mentioning that this includes checks on the behavior of `-isystem` system directories (i.e. `isystem_system_dirs`. Otherwise IMO this test would be redundant given `test_ccld_with_system_dirs ` and `test_cc_deps_isystem `.